src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -554,10 +554,10 @@ private async Task<ActionResult<MergePolicyCheckResult>> CheckMergePolicyAsync(I
             {
                 await remote.MergeDependencyPullRequestAsync(pr.Url, new MergePullRequestParameters());
             }
-            catch
+            catch (PullRequestNotMergeableException notMergeableException)
             {
-                _logger.LogInformation("NOT Merged: PR '{url}' has merge conflicts.", pr.Url);
-                return ActionResult.Create(MergePolicyCheckResult.FailedToMerge, $"NOT Merged: PR '{pr.Url}' has merge conflicts.");
+                _logger.LogInformation("NOT Merged: PR '{url}' cannot be merged. - {message}", pr.Url, notMergeableException.Message);
+                return ActionResult.Create(MergePolicyCheckResult.FailedToMerge, $"NOT Merged: PR '{pr.Url}' cannot be merged.");
```

Comment:
Here it's missing a generic `catch` block which will do `LogError(e, "...");` and return also `FailedToMerge`. Otherwise we would change the behaviour of this as we don't know where this will be caught

Reply:
added